org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalDescriptionProvider.java

```diff
@@ -331,7 +338,25 @@ private void createMethodProposalLabel(CompletionProposal methodProposal, Comple
 		setSignature(item, String.valueOf(methodProposal.getSignature()));
 		setDeclarationSignature(item, String.valueOf(methodProposal.getDeclarationSignature()));
 		setName(item, String.valueOf(methodProposal.getName()));
-
+		if (fUnit != null && methodProposal.isConstructor() && typeInfo.length() > 0 && item.getData() != null && methodProposal.getRequiredProposals() != null && methodProposal.getRequiredProposals().length > 0) {
+			CompletionProposal requiredProposal = methodProposal.getRequiredProposals()[0];
+			try {
+				Object data = item.getData();
+				if (data instanceof Map) {
+					String uri = ((Map<String, String>) data).get(CompletionResolveHandler.DATA_FIELD_URI);
```

Comment:
The latest commit works better.

A minor comment is to remove the unnecessary lines 344 ~ Line 346.